### PR TITLE
refactor(module:transfer): refactor

### PR DESCRIPTION
**Pull request by Ricbet**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #4634 

## What is the new behavior?

Add NzSelectedKeys property.  Sync antd

## Does this PR introduce a breaking change?
```
[] Yes
[x] No
```

### CODE REVIEW

File: components/transfer/transfer.spec.ts [hunk 1]
```diff
@@ -49,6 +49,19 @@ describe('transfer', () => {
...
     });
 
+    it('should be from left to right when via nzSelectedKeys property', () => {
+      instance.nzSelectedKeys = ['0', '1', '2'];
+      fixture.detectChanges();
+
+      expect(
+        instance.comp.nzSelectedKeys.every(e => {
+          const data = instance.comp.nzDataSource.find(d => d.key === e);
+          console.log(data);
```
Comment (hsuanxyz) [inline components/transfer/transfer.spec.ts:59]:
```suggestion
```

File: components/transfer/transfer.component.ts [hunk 2]
```diff
@@ -72,6 +127,7 @@ export class NzTransferComponent implements OnInit, OnChanges, OnDestroy {
   @Input() nzSearchPlaceholder: string;
   @Input() nzNotFoundContent: string;
   @Input() nzTargetKeys: string[] = [];
+  @Input() nzSelectedKeys: string[] = [];
```
Comment (hsuanxyz) [inline components/transfer/transfer.component.ts:130]: 这里需要双向绑定吗？如果不需要的话在文档里面说明一下。

Comment (Ricbet) [reply components/transfer/transfer.component.ts:130]: ok

File: components/transfer/doc/index.zh-CN.md [hunk 3]
```diff
@@ -40,6 +40,7 @@ import { NzTransferModule } from 'ng-zorro-antd/transfer';
 | `[nzSearchPlaceholder]` | 搜索框的默认值 | `string` | `'请输入搜索内容'` |
 | `[nzNotFoundContent]` | 当列表为空时显示的内容 | `string` | `'列表为空'` |
 | `[nzCanMove]` | 穿梭时二次校验。**注意：** 穿梭组件内部始终只保留一份数据，二次校验过程中需取消穿梭项则直接删除该项；具体用法见示例。 | `(arg: TransferCanMove) => Observable<TransferItem[]>` | - |
+| `[nzSelectedKeys]` | 设置被选中的 key 集合 | `string[]` | - |
```
Comment (Ricbet) [inline components/transfer/doc/index.zh-CN.md:43]: @hsuanxyz 改成这样的描述？

**Pull request merged**